lib/flake-utils.nix

```diff
@@ -0,0 +1,63 @@
+{ nixpkgs, systems }:
+rec {
+  /* eachDefaultSystem using defaultSystems
+     Usage in a nix flake:
+       {
+         description = "Flake demo";
+
+         inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+         outputs = { self, nixpkgs, flake-utils }:
+           nixpkgs.lib.eachDefaultSystem (pkgs:
+             rec {
```

Comment:
Ooh, this is tricky. We should be careful with scope. `rec` is not careful with scope.
I'd prefer a callback argument that comes from `self`. That makes the function potentially useful in other places than the root of a flake `outputs` body as well.

Reply:
Sure. It's just an example. I can change it.